src/components/data_submission/consent_group/EditConsentGroup.js

```diff
@@ -0,0 +1,249 @@
+import { div, h, h3 } from 'react-hyperscript-helpers';
+import { isNil, isString } from 'lodash/fp';
+import { FormFieldTypes, FormField, FormValidators } from '../../forms/forms';
+
+export const selectedPrimaryGroup = (consentGroup) => {
+  if (!isNil(consentGroup.generalResearchUse) && consentGroup.generalResearchUse) {
+    return {
+      selected: 'generalResearchUse',
+    };
+  } else if (!isNil(consentGroup.hmb) && consentGroup.hmb) {
+    return {
+      selected: 'hmb',
+    };
+  } else if (!isNil(consentGroup.diseaseSpecificUse) && isString(consentGroup.diseaseSpecificUse)) {
+    return {
+      selected: 'diseaseSpecificUse',
+      value: consentGroup.diseaseSpecificUse,
+    };
+  } else if (!isNil(consentGroup.poa) && consentGroup.poa) {
+    return {
+      selected: 'poa',
+    };
+  } else if (!isNil(consentGroup.otherPrimary) && isString(consentGroup.otherPrimary)) {
+    return {
+      selected: 'otherPrimary',
+      value: consentGroup.otherPrimary,
+    };
+  }
+
+  return undefined;
+};
+
+export const EditConsentGroup = (props) => {
+  const {
+    consentGroup,
+    setConsentGroup,
+    idx,
+  } = props;
+
+  const onChange = ({key, value}) => {
+    setConsentGroup({
+      ...consentGroup,
+      ...{
+        [key]: value,
+      },
+    });
+  };
+
+  const onPrimaryChange = ({key, value}) => {
+    setConsentGroup({
+      ...consentGroup,
+      ...{
+        generalResearchUse: false,
+        hmb: false,
+        diseaseSpecificUse: undefined,
+        poa: false,
+        otherPrimary: undefined,
+      },
+      ...{
+        [key]: value || true,
+      }
+    });
+  };
+
+  return div({
+    style: {
+      width: '70%'
+    }
+  }, [
+
+    h3(['Consent Group Information']),
+
+    // name
+    h(FormField, {
+      id: idx+'_consentGroupName',
+      name: 'consentGroupName',
+      title: 'Consent Group Name',
+      validators: [FormValidators.REQUIRED],
+      placeholder: 'Enter name',
+      defaultValue: consentGroup.consentGroupName,
+      onChange,
+    }),
+
+    // primary
+    h(FormField,
+      {
+        type: FormFieldTypes.RADIO,
+        id: idx+'_primaryRadio',
+        name: 'primaryRadio',
+        title: 'Consent Group - Primary Data Use Terms',
+        validators: [FormValidators.REQUIRED],
+        description: 'Please select one of the following data use permissions for your dataset',
+        options: [
+          {
+            id: 'generalResearchUse',
+            name: 'generalResearchUse',
+            text: 'General Research Use',
+          },
+          {
+            id: 'hmb',
+            name: 'hmb',
+            text: 'Health/Medical/Biomedical Research Use',
+          },
+          {
+            id: 'diseaseSpecificUse',
+            name: 'diseaseSpecificUse',
+            text: 'Disease-Specific Research Use',
+            type: 'string',
+            placeholder: 'Please enter one or more diseases',
```

Comment:
This field should function like the field on the DAR form or `DatasetRegistration.js, line 1129` in the sense that an autofill list should be popping up as you type in the input field.

Reply:
copied over functionality from the DAR form. if the user selected 'disease 1' and 'disease 1' the form will have the string: `disease 1, disease 2`, since the schema just has a string there. however, if we wanted to store the diseases as a list of strings, that would be an easy fix

Reply:
The schema is expecting an array of disease strings: https://github.com/DataBiosphere/consent/blob/6ec50f6b9ddc1649e17b1f0522107eea3084dddb/src/main/resources/dataset-registration-schema_v1.json#L305

Reply:
oh great I will make that change. makes the code easier actually